app/components/Map.client.tsx

```diff
@@ -12,42 +15,152 @@ interface Props {
       zoom: 9,
     });
 
     _map.fitBounds(bounds);
 
-    setMap(_map);
+    _map.on("load", () => {
+      _map.addSource('google-satellite', {
+        type: 'raster',
+        tiles: ['https://mt1.google.com/vt/lyrs=s&x={x}&y={y}&z={z}'],
+        tileSize: 256,
+        attribution: '© Google',
+      });
+
+      _map.addLayer({
+        id: 'google-satellite-layer',
+        type: 'raster',
+        source: 'google-satellite',
+        layout: {
+          visibility: 'none'
+        }
+      });
+
+      _map.addSource('usgs-topo', {
+        type: 'raster',
+        tiles: ['https://basemap.nationalmap.gov/arcgis/services/USGSTopo/MapServer/WMSServer?bbox={bbox-epsg-3857}&format=image/png&service=WMS&version=1.1.1&request=GetMap&srs=EPSG:3857&transparent=true&width=256&height=256&layers=0&styles=default'],
+        tileSize: 256,
+        attribution: '© USGS',
+      });
+
+      _map.addLayer({
+        id: 'usgs-topo-layer',
+        type: 'raster',
+        source: 'usgs-topo',
+        layout: {
+          visibility: 'none'
+        }
+      });
+
+      // The ATLMaps Layer (I've added it for reference on example for our geoserver)
+      _map.addSource('atlmaps', {
+        type: 'raster',
+        tiles: ['https://geoserver.ecds.emory.edu/ATLMaps/gwc/service/wms?layers=ATLMaps:r9jps&service=WMS&request=GetMap&styles=&format=image/png&transparent=true&version=1.1.1&width=256&height=256&srs=EPSG:3857&bbox={bbox-epsg-3857}'],
+        tileSize: 256,
+        attribution: '© ATLMaps',
+      });
+
+      _map.addLayer({
+        id: 'atlmaps-layer',
+        type: 'raster',
+        source: 'atlmaps',
+        layout: {
+          visibility: 'none'
+        }
+      });
+
+      setMap(_map);
+      setMapLoaded(true);
+    });
 
     return () => {
       if (_map) _map.remove();
       setMap(undefined);
       setMapLoaded(false);
     };
-  }, [setMap]);
+  }, [setMap, setMapLoaded]);
 
-  useEffect(() => {
+  const toggleLayer = (layerName: string) => {
     if (!map) return;
-    map.on("load", async () => {
-      setMapLoaded(true);
+
+    const layers = ['google-satellite-layer', 'usgs-topo-layer', 'atlmaps-layer'];
+
+    layers.forEach((layer) => {
+      map.setLayoutProperty(layer, 'visibility', 'none');
     });
-  }, [map, setMapLoaded]);
+
+    if (layerName !== 'custom') {
+      map.setLayoutProperty(`${layerName}-layer`, 'visibility', 'visible');
+    }
+
+    setActiveLayer(layerName);
+    setIsDropdownOpen(false);
+  };
 
   return (
-    <div id="mapContainer" className={`h-[calc(100vh-${topBarHeight})]`}></div>
+    <div className="relative">
+      <div ref={mapContainerRef} className={`h-[calc(100vh-${topBarHeight})]`}></div>
+      <div className="absolute top-4 right-4">
+        <button
+          onClick={() => setIsDropdownOpen(!isDropdownOpen)}
+          className="bg-white p-2 rounded-md shadow-md"
+        >
+          <FontAwesomeIcon icon={faLayerGroup} />
```

Comment:
Put this block in the new MapSwitcher component. You can pass the map object as a property or set up a [context for map ](https://react.dev/learn/passing-data-deeply-with-context)- that might be overkill though.